pkg/storaged/utils.js

```diff
@@ -236,49 +244,69 @@ export function get_block_link_parts(client, path) {
     if (!block)
         return;
 
-    let location, link;
+    let location, link, name, subname;
     if (client.mdraids[block.MDRaid]) {
         location = ["mdraid", client.mdraids[block.MDRaid].UUID];
         link = cockpit.format(_("RAID device $0"), mdraid_name(client.mdraids[block.MDRaid]));
     } else if (client.blocks_lvm2[path] &&
                client.lvols[client.blocks_lvm2[path].LogicalVolume] &&
                client.vgroups[client.lvols[client.blocks_lvm2[path].LogicalVolume].VolumeGroup]) {
-        const target = client.vgroups[client.lvols[client.blocks_lvm2[path].LogicalVolume].VolumeGroup].Name;
+        const lvol = client.lvols[client.blocks_lvm2[path].LogicalVolume];
+        const target = client.vgroups[lvol.VolumeGroup].Name;
         location = ["vg", target];
         link = cockpit.format(_("LVM2 volume group $0"), target);
+        name = target;
+        subname = lvol.Name;
     } else {
         const vdo = client.legacy_vdo_overlay.find_by_block(block);
         if (vdo) {
             location = ["vdo", vdo.name];
             link = cockpit.format(_("VDO device $0"), vdo.name);
         } else {
             location = [block_name(block).replace(/^\/dev\//, "")];
-            if (client.drives[block.Drive])
+            if (client.drives[block.Drive]) {
                 link = drive_name(client.drives[block.Drive]);
-            else
+                name = drive_main_name(client.drives[block.Drive]);
+                subname = drive_sub_name(client.drives[block.Drive]);
+            } else
                 link = block_name(block);
         }
     }
 
+    if (!name) {
+        name = link;
+        subname = "";
+    }
+
     // Partitions of logical volumes are shown as just logical volumes.
-    let format;
-    if (is_lvol && is_crypt)
+    let format, type;
+    if (is_lvol && is_crypt) {
         format = _("Encrypted logical volume of $0");
-    else if (is_part && is_crypt)
+        type = _("logical volume");
+    } else if (is_part && is_crypt) {
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18967-20230630-124833-368c0200-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)